git_workflow_quick_reference.md

```diff
@@ -0,0 +1,159 @@
+
+### Push your changes to the remote
+
+Pushing your changes lets other developers access them, and also backs them up in the remote repository. Pushing regularly is good practice, as it keeps others in the loop and prevents you from losing work. To push your branch:
+
+```
+git push origin phil/JIRA-1234/profile-page
+```
+
+Note: If this command fails with a message "(TODO: Fill in this message.)" then that means that your local branch has diverged from a version previously pushed to the server. This may happen after a rebase. Read *this* for more on how to work around this issue.
+
+### Post your changes for review
+
+Once your branch is pushed, you should notifiy other developers to review it.
+- Create a pull request on github, comparing your branch to the main branch. (See our *pull request guide* for specifics on how to create a good pull request.)
+- Ask another developer to review your PR. Some teams may choose to use github's "assign" feature to assign a reviewer.
+
+When reviewing someone else's changes:
+- Look over the code and leave comments, or talk to them in person.
+- If you have time, pull and run the feature branch to get a good understanding of what it does. This can sometimes be useful for gaining context before reading the code.
+- After you are satisfied with the code, leave a "Ship it!" message. (TODO: Link various shipit squirrels, boats, thumbs-ups, etc.)
+
+### Merging changes into the main branch
+
+After your pull request has been reviewed and recieved a "ship it", it's time to merge. First check whether the main branch has been updated since the time that you created the pull request.
+
+(TODO: Rewrite this next section to be more fool-proof, and to explain how to determine if you branch needs to be rebased again.)
+
+If your branch needs to be rebased, do the following:
+```
+git checkout develop
+git pull origin develop
+git checkout phil/JIRA-1234/profile-page
+git rebase -i develop
+```
+
+Now push the updated branch to the server.
+```
+git push -f origin phil/JIRA-1234/profile-page
+```
+Warning: Note that the `-f` flag is to "force" git to push the updated branch. Because `rebase` rewrites history, git will not allow you to push the new version of the branch by default. You should *never* use `-f` except when updating your own feature branches.
+
+Revisit your pull request on github. There should be a green "merge" button telling you that your pull request can be automatically merged. Press this button to merge your changes into the remote branch.
+
+TODO: Should we move away from using github's merge button, and instead merge branches locally? The advantages would be:
+- It reduces the need to push with `-f`.
+- It ensures that history remains linear, instead of possible edge-cases where two developers merge at the same time.
```

Comment:
I like having a merge commit at the end of each PR. This way, it's very easy to tell which commit belongs to which PR and where each PR begins/ends. The Github merge button does this automatically. This can also be done locally but needs a bit more configuration. Either way, the merge commit should just contain a message as separator and no actual changes.